includes/class-wcpdf-admin.php

```diff
@@ -1038,8 +1038,8 @@ public function ajax_crud_document() {
 		);
 
 		try {
-			// Allow getting documents without invoice.
-			add_filter( 'wpo_wcpdf_require_invoice_documents', '__return_empty_array()' );
+			// Allow getting documents even if their prerequisite documents doesn't exist.
+			add_filter( 'wpo_wcpdf_prerequisite_documents', '__return_empty_array()' );
```

Comment:
Maybe here you can call the function?

Reply:
No, calling the function here or another function with the code you mentioned below that was in Pro. I don't think you need to set the array as empty because it's an empty array property already?

Reply:
> I don't think you need to set the array as empty because it's an empty array property already?

I think it might not be empty. For example, for the receipt document when "Require invoice" is checked, the prerequisite documents is not empty, and if we don't make it empty, the issue you mentioned [here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/757#issuecomment-2051309553) could happen after removing the invoice.

What function do you think should be called here?

Reply:
I think you can call the `is_allowed()` and include a new `$context` arg in it? This way you can create a filter inside it to allow or not in different contexts.

Reply:
The `wcpdf_get_document()` method, which will be called in the next lines to get the document, will call the `is_allowed()` method again separately, so I'm not sure if calling the `is_allowed()` before that could help in this implementation.

Since the `is_allowed()` method checks for different conditions and here, we just needed to modify prerequisite documents, I thought maybe applying this filter could be good. But if modifying prerequisites using the filter is not preferred, then as you mentioned, we can add a new `$context` arg to the `is_allowed()` method and a new filter inside it, So we can use that filter here before the `wcpdf_get_document()` to change the result of `is_allowed()`.

What do you think?

Reply:
I suggested previously to create a separated function for this check: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/757/commits/48d03715bf5e586a79098d22eb181d9045ef5c95#diff-b4b45c08877ccc6f215e534426a81b3edd82ce371cfc3fce7420e62e279b3c1eR538-R545

This way you don't need to rely on `is_allowed()` I think?

Reply:
I think you're right.

Reply:
I created a new method, `do_prerequisites_exist(),` but this method still needs to be used inside `wcpdf_get_document()` to prevent the generation of a document if prerequisites do not exist. So, again, this approach ended up with adding a filter to change the `do_prerequisites_exist()` method value to allow displaying documents in meta box.

Please let me know what other solution you suggest.